internal/datasources/service/convert.go

```diff
@@ -60,3 +62,26 @@ func dataSourceRestDBToProtobuf(ds *minderv1.DataSource, dsfuncs []db.DataSource
 
 	return ds, nil
 }
+
+func dataSourceStructDBToProtobuf(ds *minderv1.DataSource, dsfuncs []db.DataSourcesFunction) (*minderv1.DataSource, error) {
+	// At this point we have already validated that we have at least one function.
+	ds.Driver = &minderv1.DataSource_Structured{
+		Structured: &minderv1.StructDataSource{
+			Def: make(map[string]*minderv1.StructDataSource_Def, len(dsfuncs)),
+		},
+	}
+
+	for _, dsf := range dsfuncs {
+		key := dsf.Name
+		dsfToParse := &minderv1.StructDataSource_Def{
+			Path: &minderv1.StructDataSource_Def_Path{},
+		}
+		if err := protojson.Unmarshal(dsf.Definition, dsfToParse); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal data source definition for %s: %w", key, err)
+		}
+
+		ds.GetStructured().Def[key] = dsfToParse
```

Comment:
this looks panicky. Should we verify that `GetStructured` isn't nil?

Reply:
Can't be nil, it's defined on line 69 (it's analogous to the REST equivalent on L60)

Reply:
I refactored the func to make it clearer, PTAL!